File: LandingBack/Data/Modelos/Propiedad.cs
using System.ComponentModel.DataAnnotations.Schema;
// using System.Drawing; // Comentado temporalmente
// using System.Text.Json;

namespace LandingBack.Data.Modelos
{
    public class Propiedad
    {
        public int Id { get; set; }
        public string Codigo { get; set; } = null!;
        public string Tipo { get; set; } = null!;        // Departamento|Casa|PH|...
        public string Operacion { get; set; } = "Venta"; // Venta|Alquiler
        public string Barrio { get; set; } = null!;
        public string Comuna { get; set; } = null!;
        public string Direccion { get; set; } = null!;
        // Coordenadas geográficas como campos separados para Entity Framework
        public double? GeoLatitud { get; set; }
        public double? GeoLongitud { get; set; }
        public string Moneda { get; set; } = "USD";      // USD|ARS
        public decimal Precio { get; set; }
        public decimal? Expensas { get; set; }
        public int Ambientes { get; set; }
        public int? Dormitorios { get; set; }
        public int? Banos { get; set; }
        public bool? Cochera { get; set; }
        public int? MetrosCubiertos { get; set; }
        public int? MetrosTotales { get; set; }
        public int? Antiguedad { get; set; }
        public int? Piso { get; set; }
        public bool? AptoCredito { get; set; }
        public string? AmenitiesJson { get; set; }     // Para almacenar JSON como string en la DB
        public string Estado { get; set; } = "Activo";   // Activa|Reservada|Vendida|Pausada
        public bool Destacado { get; set; } = false;
        public DateTime FechaPublicacionUtc { get; set; } = DateTime.UtcNow;
        
        // Propiedades adicionales para búsqueda avanzada
        public string? Titulo { get; set; }
        public string? Descripcion { get; set; }
        public string? Localidad { get; set; }
        public string? Provincia { get; set; }
        public string? CodigoPostal { get; set; }
        public double? Latitud { get; set; }
        public double? Longitud { get; set; }
        public string? Orientacion { get; set; }
        public bool DisponibilidadInmediata { get; set; } = true;
        public bool AceptaMascotas { get; set; } = false;
        public bool AceptaCredito { get; set; } = false;
        public DateTime FechaCreacion { get; set; } = DateTime.UtcNow;
        public DateTime FechaActualizacion { get; set; } = DateTime.UtcNow;
        
        // Aliases para compatibilidad con búsquedas
        public string TipoPropiedad => Tipo;
        public string TipoOperacion => Operacion;
        public int? Baños => Banos;
        public int? SuperficieCubierta => MetrosCubiertos;
        public int? SuperficieTotal => MetrosTotales;

        // Navegación
        public ICollection<PropiedadMedia> Medias { get; set; } = new List<PropiedadMedia>();
        public ICollection<PropiedadMedia> PropiedadMedias => Medias; // Alias para compatibilidad
        public ICollection<PropiedadHistorial> Historial { get; set; } = new List<PropiedadHistorial>();
    }
}

File: frontend/src/types/propiedad.ts
// Tipos TypeScript para el módulo de propiedades

export interface Propiedad {
  id: number;
  codigo: string;
  tipo: string;
  operacion: 'Venta' | 'Alquiler';
  barrio: string;
  comuna: string;
  direccion: string;
  latitud?: number | null;
  longitud?: number | null;
  moneda: 'USD' | 'ARS';
  precio: number;
  expensas?: number | null;
  ambientes: number;
  dormitorios?: number | null;
  banos?: number | null;
  cochera?: boolean;
  metrosCubiertos?: number | null;
  metrosTotales?: number | null;
  antiguedad?: number | null;
  piso?: number | null;
  aptoCredito?: boolean;
  estado: 'Activo' | 'Reservado' | 'Vendido' | 'Pausado';
  destacado: boolean;
  titulo?: string;
  descripcion?: string;
  amenities?: Record<string, boolean>;
  fechaPublicacionUtc: string;
  fechaCreacion: string;
  fechaActualizacion: string;
  medias?: PropiedadMedia[];
}

export interface PropiedadMedia {
  id: number;
  propiedadId: number;
  url: string;
  titulo?: string;
  tipo: 'image' | 'video' | 'tour';
  tipoArchivo: string;
  orden: number;
  esPrincipal: boolean;
}

export interface PropiedadCreateDto {
  codigo: string;
  tipo: string;
  operacion: 'Venta' | 'Alquiler';
  barrio: string;
  comuna: string;
  direccion: string;
  latitud?: number;
  longitud?: number;
  moneda: 'USD' | 'ARS';
  precio: number;
  expensas?: number;
  ambientes: number;
  dormitorios?: number;
  banos?: number;
  cochera?: boolean;
  metrosCubiertos?: number;
  metrosTotales?: number;
  antiguedad?: number;
  piso?: number;
  aptoCredito?: boolean;
  estado: 'Activo' | 'Reservado' | 'Vendido' | 'Pausado';
  destacado: boolean;
  titulo?: string;
  descripcion?: string;
  amenities?: Record<string, boolean>;
}

export interface PropiedadUpdateDto extends PropiedadCreateDto {
  id: number;
}

export interface PropiedadFilters {
  operacion?: string;
  tipo?: string;
  barrio?: string;
  comuna?: string;
  precioMin?: number;
  precioMax?: number;
  ambientes?: number;
  dormitorios?: number;
  cochera?: boolean;
  estado?: string;
  destacado?: boolean;
  page: number;
  pageSize: number;
  orderBy: string;
  orderDesc: boolean;
  searchTerm?: string;
}

export interface PaginationInfo {
  totalCount: number;
  totalPages: number;
  currentPage: number;
  pageSize: number;
}

export interface MediaUploadData {
  propiedadId: number;
  file: File;
  titulo?: string;
  orden?: number;
}

export interface MediaUrlData {
  propiedadId: number;
  url: string;
  titulo?: string;
  tipo: 'image' | 'video' | 'tour';
  orden?: number;
}

export interface UrlValidationResult {
  valid: boolean;
  errors?: string[];
  detectedType?: string;
  isSupported?: boolean;
  processedUrl?: string;
  message?: string;
}
